atlassian/bitbucket/cloud/repositories/pullRequests.py

```diff
@@ -0,0 +1,248 @@
+# coding=utf-8
+
+from ..base import BitbucketCloudBase
+from ..common.users import User
+from ..const import (
+    PRP_APPROVED,
+    PRP_CHANGES_REQUESTED,
+    PRP_ROLE_PARTICIPANT,
+    PRP_ROLE_REVIEWER,
+    PR_MERGE_COMMIT,
+    PR_MERGE_STRATEGIES,
+    PR_STATE_DECLINED,
+    PR_STATE_MERGED,
+    PR_STATE_OPEN,
+    PR_STATE_SUPERSEDED,
+)
+
+
+class PullRequests(BitbucketCloudBase):
+    def __init__(self, url, *args, **kwargs):
+        super(PullRequests, self).__init__(url, *args, **kwargs)
+
+    def __get_object(self, data):
+        if "errors" in data:
+            return
+        return PullRequest(self.url_joiner(self.url, data["id"]), data, **self._new_session_args)
+
+    def each(self, q=None, sort=None):
+        """
+        Returns the list of pull requests in this repository.
+
+        :param q: string: Query string to narrow down the response.
+                          See https://developer.atlassian.com/bitbucket/api/2/reference/meta/filtering for details.
+        :param sort: string: Name of a response property to sort results.
+                             See https://developer.atlassian.com/bitbucket/api/2/reference/meta/filtering for details.
+
+        :return: A generator for the PullRequest objects
+        """
+        params = {}
+        if sort is not None:
+            params["sort"] = sort
+        if q is not None:
+            params["q"] = q
+        for pr in self._get_paged(None, trailing=True, params=params):
+            yield self.__get_object(pr)
+
+        return
+
+    def get(self, id):
+        """
+        Returns the pull requests with the requested id in this repository.
+
+        :param id: int: The requested pull request id
+
+        :return: The requested PullRequest object
+        """
+        return self.__get_object(super(PullRequests, self).get(id))
+
+
+class PullRequest(BitbucketCloudBase):
+    def __init__(self, url, data, *args, **kwargs):
+        super(PullRequest, self).__init__(url, *args, data=data, expected_type="pullrequest", **kwargs)
+
+    def _check_if_open(self):
+        if not self.is_open:
+            raise Exception("Pull Request isn't open")
+        return
+
+    @property
+    def id(self):
+        """ unique pull request id """
+        return self.get_data("id")
+
+    @property
+    def title(self):
+        """ pull request title """
+        return self.get_data("title")
+
+    @property
+    def description(self):
+        """ pull request description """
+        return self.get_data("description")
+
+    @property
+    def is_declined(self):
+        """ True if the pull request was declined """
+        return self.get_data("state").upper() == PR_STATE_DECLINED
```

Comment:
The `upper()` isn't needed because the constants are documented in the API with a exact case. This MUST match.